### PR TITLE
Add Canopy and Post-Halving test vectors

### LINKED ISSUE #1099: Add Mainnet Canopy Block Test Vectors
In #1091, we added block test vectors for each network upgrade transition.

The next network upgrade is Canopy on Mainnet. We want to add Mainnet Canopy test vectors after Canopy activates - around **18 November 2020**. This is also the first Mainnet block subsidy halving.

Preconditions:
- [x] Canopy has activated on Mainnet (around 18 November 2020)
- [x] Mainnet is past the block reorg limit for the Canopy activation (that is, 1_028_602 or later)

Software and Data Requirements:
- [x] A syncronised zcashd mainnet instance
- [x] The `zcash-cli` RPC tool

Tasks:
- [x] Add the hex-encoded data of mainnet blocks 1_028_499, 1_028_500, and 1_028_501 to https://github.com/ZcashFoundation/zebra/tree/main/zebra-test/src/vectors/
- [x] Add the hex-encoded data of the highest finalized mainnet block to https://github.com/ZcashFoundation/zebra/tree/main/zebra-test/src/vectors/
  - Note: finalized blocks must be at least 100 blocks behind the tip height
- [x] Add lazy_static refs for each of those blocks to https://github.com/ZcashFoundation/zebra/tree/main/zebra-test/src/vectors/block.rs
- [x] Add the block refs to the mainnet list at https://github.com/ZcashFoundation/zebra/tree/main/zebra-test/src/vectors/block.rs
- [x] Run the Zebra tests, and check that they pass

### LINKED ISSUE #1104: Add Testnet Post-Halving Block Test Vectors
In #1091, we added block test vectors for each network upgrade transition.

The Testnet block subsidy halves on Testnet at block 1_116_000. We want to add post-halving Testnet test vectors around **13 October 2020**, after the first halving. (The first Mainnet halving is at Canopy activation, so it is covered by #1099.)

Preconditions:
- [x] The block subsidy has halved on Testnet (around 13 October 2020)
- [x] Testnet is past the block reorg limit for the first halving (that is, 1_116_102 or later)

Software and Data Requirements:
- [x] A syncronised zcashd testnet instance
- [x] The `zcash-cli` RPC tool

Tasks:
- [x] Add the hex-encoded data of testnet blocks 1_115_999, 1_116_000, and 1_116_001 to https://github.com/ZcashFoundation/zebra/tree/main/zebra-test/src/vectors/
- [x] Add the hex-encoded data of the highest finalized testnet block to https://github.com/ZcashFoundation/zebra/tree/main/zebra-test/src/vectors/
  - Note: finalized blocks must be at least 100 blocks behind the tip height
- [x] Add lazy_static refs for each of those blocks to https://github.com/ZcashFoundation/zebra/tree/main/zebra-test/src/vectors/block.rs
- [x] Add the block refs to the testnet block list in https://github.com/ZcashFoundation/zebra/tree/main/zebra-test/src/vectors/block.rs
- [x] Run the Zebra tests, and check that they pass

### PR DESCRIPTION
**Pull request by teor2345**
## Motivation

We want to test Post-Canopy blocks on mainnet, and Post-Halving blocks on testnet.

## Solution

- Add post-canopy mainnet block test vectors
- Add post-halving testnet block test vectors

The code in this pull request has:
  - [x] Documentation Comments
  - [x] Existing Unit Tests and Property Tests

## Review

@oxarbitrage wants this change for #1903

### Review Note

GitHub displays `zebra-test/src/vectors/block-main-1-046-401.bin` as an empty file in the unified diff view, but it actually contains 72 kB of binary data.

## Related Issues

Closes #1099 
Closes #1104

## Follow Up Work

We'll need to do something similar for NU5.

### CODE REVIEW

File: zebra-test/src/vectors/block.rs [hunk 1]
```diff
@@ -276,14 +286,29 @@ lazy_static! {
...
-    // TODO: one more Canopy Mainnet block
-    //       (so that we have at least 3 blocks from Canopy)
-    // Note: don't use the highest block, it must be below the reorg limit!
+    pub static ref BLOCK_MAINNET_1046399_BYTES: Vec<u8> =
+        <Vec<u8>>::from_hex(include_str!("block-main-1-046-399.txt").trim())
+            .expect("Block bytes are in valid hex representation");
+    pub static ref BLOCK_MAINNET_1046400_BYTES: Vec<u8> =
+        <Vec<u8>>::from_hex(include_str!("block-main-1-046-400.txt").trim())
+            .expect("Block bytes are in valid hex representation");
+    // Block 1046401 is 143kB in size (currently the second-largest test vector), so we store it in binary.
```
Comment (teor2345) [inline zebra-test/src/vectors/block.rs]:
```suggestion
    // Block 1046401 is 72kB in size (currently the second-largest test vector), so we store it in binary.
```

**Comment by teor2345** [comment]
@oxarbitrage just a note for your review:

GitHub displays `zebra-test/src/vectors/block-main-1-046-401.bin` as an empty file in the unified diff view, but it actually contains 72 kB of binary data.

**Review by oxarbitrage — APPROVED** [review]
Thank you @teor2345 , looks great.

**Review by dconnolly — APPROVED** [review]
🎉📦🔗

**Pull request merged**